compiler/front/msgs.nim

```diff
@@ -654,6 +656,26 @@ template globalReport*(conf: ConfigRef, report: ReportTypes) =
   handleReport(
     conf, wrap(report, instLoc()), instLoc(), doRaise)
 
+proc reportAndForceRaise*(
+  conf: ConfigRef, r: Report, reportFrom: InstantiationInfo) =
+  ## always call `raiseRecoverableError` except if `doAbort` was returned
+  ## by the `structuredReportHook`.
```

Comment:
```suggestion
  ## Similar to `handleReport`, but, unless overridden with aborting
  ## (`doAbort`) by the structured report hook, always raises a recoverable
  ## error.
```
This focuses more on the concrete behaviour, rather than on the implementation.